controllers/api_controller_integration_test.go

```diff
@@ -914,166 +765,151 @@ var _ = Describe("APIRule Controller", func() {
-					err = c.Get(context.TODO(), client.ObjectKey{Name: apiRuleName, Namespace: testNamespace}, &expectedApiRule)
-					Expect(err).NotTo(HaveOccurred())
-
-					Expect(expectedApiRule.Status.APIRuleStatus.Code).To(Equal(gatewayv1beta1.StatusOK))
 				})
 			})
 		})
 	})
-})
 
-func add(mgr manager.Manager, r reconcile.Reconciler) error {
-	c, err := controller.New("api-gateway-controller", mgr, controller.Options{Reconciler: r})
-	if err != nil {
-		return err
-	}
-
-	err = c.Watch(&source.Kind{Type: &gatewayv1beta1.APIRule{}}, &handler.EnqueueRequestForObject{})
-	if err != nil {
-		return err
-	}
+	It("APIRule in status Error should reconcile to status OK when root cause of error is fixed", func() {
+		// given
+		updateJwtHandlerTo(helpers.JWT_HANDLER_ISTIO)
 
-	err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForObject{})
-	if err != nil {
-		return err
-	}
+		apiRuleName := generateTestName(testNameBase, testIDLength)
+		serviceName := generateTestName(testServiceNameBase, testIDLength)
+		serviceHost := fmt.Sprintf("%s.kyma.local", serviceName)
+		vsName := generateTestName("duplicated-host-vs", testIDLength)
 
-	return nil
-}
+		By(fmt.Sprintf("Create Virtual service for host %s", serviceHost))
+		vs := virtualService(vsName, serviceHost)
+		Expect(c.Create(context.TODO(), vs)).Should(Succeed())
+		defer func() {
+			By(fmt.Sprintf("Deleting VirtualService %s as part of teardown", vs.Name))
+			Eventually(func(g Gomega) {
+				_ = c.Delete(context.TODO(), vs)
+				v := networkingv1beta1.VirtualService{}
+				err := c.Get(context.TODO(), client.ObjectKey{Name: vs.Name, Namespace: testNamespace}, &v)
+				g.Expect(errors.IsNotFound(err)).To(BeTrue())
+			}, eventuallyTimeout).Should(Succeed())
+		}()
+
+		By("Verify VirtualService is created")
+		Eventually(func(g Gomega) {
+			createdVs := networkingv1beta1.VirtualService{}
+			g.Expect(c.Get(context.TODO(), client.ObjectKey{Name: vsName, Namespace: testNamespace}, &createdVs)).Should(Succeed())
+		}, eventuallyTimeout).Should(Succeed())
+
+		apiRuleLabelMatcher := matchingLabelsFunc(apiRuleName, testNamespace)
+
+		By("Create APIRule")
+		rule := testRule("/headers", []string{"GET"}, nil, testIstioJWTHandler(testIssuer, testJwksUri))
+		instance := testInstance(apiRuleName, testNamespace, serviceName, serviceHost, testServicePort, []gatewayv1beta1.Rule{rule})
+
+		Expect(c.Create(context.TODO(), instance)).Should(Succeed())
+		defer func() {
+			deleteApiRule(instance)
+		}()
+
+		expectApiRuleStatus(apiRuleName, gatewayv1beta1.StatusError)
+
+		By("Verify VirtualService for APIRule is not created")
+		Eventually(func(g Gomega) {
+			vsList := networkingv1beta1.VirtualServiceList{}
+			g.Expect(c.List(context.TODO(), &vsList, apiRuleLabelMatcher)).Should(Succeed())
+			g.Expect(vsList.Items).To(HaveLen(0))
+		}, eventuallyTimeout).Should(Succeed())
+
+		By("Deleting existing VirtualService with duplicated host configuration")
```

Comment:
This commenting grammar convention is inconsistent, we should either have every `By` in infinitive `-ing` form or in imperative `Verify`, `Create`